backend/terraform/0-provider.tf

```diff
@@ -0,0 +1,22 @@
+# Project dependencies
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+    random = {
+      source  = "hashicorp/random"
```

Comment:
What is this being used for?

Reply:
I think its for generating a random string for the s3 bucket. Might not be necessary.

Reply:
Ahh I saw that, interesting. Thanks!

Reply:
```suggestion
      # Generates a random string for the S3 bucket
      source  = "hashicorp/random"
```